src/state/apis/fiatRamps/fiatRamps.ts

```diff
@@ -1,54 +1,67 @@
-    [k in FiatRampAction]: FiatRampAsset[]
+export type FiatRampsByAction = {
+  [FiatRampAction.Buy]: FiatRamp[]
+  [FiatRampAction.Sell]: FiatRamp[]
+}
+
+export type FiatRampsByAssetId = {
+  byAssetId: {
+    [k: AssetId]: FiatRampsByAction | undefined
   }
+  buyAssetIds: AssetId[]
+  sellAssetIds: AssetId[]
 }
 
 export const fiatRampApi = createApi({
   ...BASE_RTK_CREATE_API_CONFIG,
   reducerPath: 'fiatRampApi',
   endpoints: build => ({
-    getFiatRampAssets: build.query<FiatRampApiReturn, void>({
+    getFiatRamps: build.query<FiatRampsByAssetId, void>({
+      keepUnusedDataFor: Number.MAX_SAFE_INTEGER, // never refetch these
       queryFn: async () => {
         try {
           const promiseResults = await Promise.allSettled(
-            Object.values(supportedFiatRamps)
-              .filter(provider => provider.isImplemented)
-              .map(provider => provider.getBuyAndSellList()),
+            Object.values(supportedFiatRamps).map(provider => provider.getBuyAndSellList()),
           )
 
-          const initial = fiatRamps.reduce<FiatRampApiReturn>((acc, cur) => {
-            acc[cur] = { [FiatRampAction.Buy]: [], [FiatRampAction.Sell]: [] }
-            return acc
-          }, {} as FiatRampApiReturn)
-
-          const data = promiseResults.reduce<FiatRampApiReturn>((acc, p, idx) => {
-            if (p.status === 'rejected') {
-              moduleLogger.error(p.reason, 'error fetching fiat ramp')
+          const data = promiseResults.reduce<FiatRampsByAssetId>(
+            (acc, p, idx) => {
+              if (p.status === 'rejected') {
+                moduleLogger.error(p.reason, 'error fetching fiat ramp')
+                return acc
+              }
+              const ramp = p.value
+              const [buyAssetIds, sellAssetIds] = ramp
+              buyAssetIds.forEach(assetId => {
+                if (!acc.byAssetId[assetId]) acc.byAssetId[assetId] = { buy: [], sell: [] }
+                acc.byAssetId[assetId]?.['buy'].push(fiatRamps[idx])
+                if (!acc.buyAssetIds.includes(assetId)) acc.buyAssetIds.push(assetId)
+              })
+              sellAssetIds.forEach(assetId => {
+                if (!acc.byAssetId[assetId]) acc.byAssetId[assetId] = { buy: [], sell: [] }
+                acc.byAssetId[assetId]?.['sell'].push(fiatRamps[idx])
+                if (!acc.sellAssetIds.includes(assetId)) acc.sellAssetIds.push(assetId)
+              })
               return acc
-            }
-            const ramp = p.value
-            const [buyAssets, sellAssets] = ramp
-            acc[fiatRamps[idx]][FiatRampAction.Buy].push(...buyAssets)
-            acc[fiatRamps[idx]][FiatRampAction.Sell].push(...sellAssets)
-            return acc
-          }, initial)
+            },
+            { byAssetId: {}, buyAssetIds: [], sellAssetIds: [] },
+          )
           return { data }
         } catch (e) {
+          const error = 'getFiatRampAssets: error fetching fiat ramp(s)'
```

Comment:
we should probably follow this pattern in all our rtk apis, otherwise this swallows silently

Reply:
Ab-solutely, been noticing I had to log errors that were swallowed by a RTK query catch recently, we will need to have a pass through all of them to make sure caught errors are properly re-logged

Reply:
we can probably actually middleware this but might be overkill, just something to be aware of